mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -206,14 +192,18 @@ public MicoApplication addMicoServiceToMicoApplicationByShortNameAndVersion(Stri
         } else if (micoServices.size() == 0) {
             // Service not included yet, simply add it
             MicoServiceDeploymentInfo micoServiceDeploymentInfo = new MicoServiceDeploymentInfo().setService(micoService);
-            setDefaultEnvironmentVariablesForKafkaEnabledService(micoServiceDeploymentInfo);
 
             // Both the service list and the service deployment info list of the application need to be updated ...
             micoApplication.getServices().add(micoService);
             micoApplication.getServiceDeploymentInfos().add(micoServiceDeploymentInfo);
 
             // ... before the application can be saved.
-            return applicationRepository.save(micoApplication);
+            applicationRepository.save(micoApplication);
+
+            // Set default deployment information (environment variables, topics)
+            serviceDeploymentInfoBroker.setDefaultDeploymentInformationForKafkaEnabledService(micoServiceDeploymentInfo);
+            serviceDeploymentInfoBroker.updateMicoServiceDeploymentInformation(applicationShortName, applicationVersion, serviceShortName,
+                new MicoServiceDeploymentInfoRequestDTO(micoServiceDeploymentInfo));
```

Comment:
Yes I know. That was a design decision made together with Jakob.
In this special case it is only possible to unpack the DTO inside the broker, because we apply the values of the DTO onto an existing object stored in the database. Only the broker is able to get this existing object from the database.

Reply:
Ok